src/components/Input/Input.jsx

```diff
@@ -0,0 +1,178 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { v4 as uuid } from 'uuid';
+import Cleave from 'cleave.js/react';
+import * as InputMasks from './InputMasks';
+import './Input.scss';
+
+const Input = ({
+  autoComplete,
+  autoFocus,
+  className,
+  id,
+  inputMask,
+  isDisabled,
+  isRequired,
+  label,
+  name,
+  onBlur,
+  onChange,
+  onFocus,
+  placeholder,
+  type,
+  value,
+}) => {
+
+  const handleChange = (e) => {
+    onChange(e);
+  };
+
+  const handleFocus = (e) => {
+    if (onFocus) onFocus(e);
+    e.currentTarget.select();
+  };
+
+  const handleBlur = (e) => {
+    if (onBlur) onBlur(e);
+  };
+
+  const inputClasses = classNames(
+    className,
+    'input',
+    {
+      // [`font-size-${size}`]: size,
+      // [`font-color-${color}`]: color,
+    }
+  );
+
+  const inputId = id || uuid();
+
+  const getInputMask = () => {
+    if (!inputMask) return undefined;
+
+    if (typeof inputMask === 'string') {
+      return InputMasks[inputMask];
+    } else {
+      return inputMask;
+    }
+  };
+
+  const renderInput = () => {
+    const inputProps = {
+      autoComplete,
+      autoFocus,
+      className: inputClasses,
+      disabled: isDisabled,
+      id: inputId,
+      name,
+      onBlur: handleBlur,
+      onChange: handleChange,
+      onFocus: handleFocus,
+      placeholder,
+      type,
+      value,
+    };
+
+    return !getInputMask() ? <input {...inputProps} /> : <Cleave {...inputProps} options={getInputMask()} />;
```

Comment:
At a higher level, I'm wondering if `TextInput` should be the name of this component. Also wondering if PhoneInput or CreditCardInput should be their own components? If we can inherit from TextInput efficiently, I believe this would reduce cognitive complexity for the consumer.

At a lower level, should we be checking the `inputMask` prop on this line? And once again, we don't need to check a value in `getInputMask()` because it won't be able to fire this function unless `inputMask` is truthy.

Reply:
I will definitely rename this to TextInput. I like it much better. In terms of having a PhoneInput, and a CreditCard input, I'm still not sold on them reducing cognitive load more than the inputMask, since we're allowing simple string values for the masks. They would be super easy to abstract though so we can just ask the team and go from there. my feeling is that this has the potential to grow to a list of components that share the exact same API but do only very slightly different things, in which case props are doing their job properly.